src/main/java/Duke.java

```diff
@@ -1,10 +1,106 @@
+import java.util.ArrayList;
+import java.util.Scanner;
 public class Duke {
+    private final UI ui = new UI();
+    private final Actions actionList = new Actions();
+    private final Parser parser = new Parser();
+    private final Save savior = new Save();
+
+    public Duke() {
+        ArrayList<Task> loadedTasks = savior.loadTasks();
+        actionList.add(loadedTasks);
+    }
+
+    public void initiateChatbot() {
+        ui.openingMessage();
+        String input = ui.readInput();
+        Parser.Command command = parser.getCommand(input);
+        while (command != Parser.Command.BYE) {
+            try {
+                String[] inputParts = parser.splitInput(input);
+                switch (command) {
+                case LIST:
+                    ui.lineSandwich(actionList.stringList());
+                    break;
+                case MARK:
```

Comment:
You should consider adding a comment to explicitly say this case has a fallthrough.